### PR TITLE
OpenID4VP: Use constants and add links to their godoc

### PR DESCRIPTION
**Pull request by reinkrul**

### CODE REVIEW

File: auth/api/iam/types.go [hunk 1]
```diff
@@ -45,18 +45,15 @@ const (
 
 var responseTypesSupported = []responseType{responseTypeCode, responseTypeVPToken, responseTypeVPIDToken}
 
-// responseMode
-type responseMode = string
-
```
Comment (gerardsn) [inline auth/api/iam/types.go:50]: `responseType` and `grantType` should then also be changed

File: auth/api/iam/types.go [hunk 2]
```diff
@@ -99,6 +96,58 @@ var vpFormatsSupported = map[string]map[string][]string{
 // https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-verifier-metadata-managemen
 var clientIdSchemesSupported = []string{"did"}
 
+// clientIDParam is the name of the client_id parameter.
+// Specified by https://datatracker.ietf.org/doc/html/rfc6749#section-2.2
+const clientIDParam = "client_id"
+
+// responseTypeParam is the name of the response_type parameter.
+// Specified by https://datatracker.ietf.org/doc/html/rfc6749#section-3.1.1
+const responseTypeParam = "response_type"
```
Comment (gerardsn) [inline auth/api/iam/types.go:105]: related to `responseType`

File: auth/api/iam/types.go [hunk 3]
```diff
@@ -99,6 +96,58 @@ var vpFormatsSupported = map[string]map[string][]string{
...
+// Specified by https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-authorization-request
+const clientMetadataURIParam = "client_metadata_uri"
+
+// clientIDSchemeParam is the name of the OpenID4VP client_id_scheme parameter.
+// Specified by https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-authorization-request
+const clientIDSchemeParam = "client_id_scheme"
+
+// vpTokenParam is the name of the OpenID4VP vp_token parameter.
+// Specified by https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-response-type-vp_token
+const vpTokenParam = "vp_token"
```
Comment (gerardsn) [inline auth/api/iam/types.go:145]: also declared as `responseTypeVPToken` and `grantTypeVPToken`. It is correct, but feels silly to have 3 constants for the same value

File: auth/api/iam/types.go [hunk 4]
```diff
@@ -99,6 +96,58 @@ var vpFormatsSupported = map[string]map[string][]string{
...
+// Specified by https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-authorization-request
+const clientMetadataParam = "client_metadata"
+
+// clientMetadataParam is the name of the OpenID4VP client_metadata_uri parameter.
+// Specified by https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-authorization-request
+const clientMetadataURIParam = "client_metadata_uri"
+
+// clientIDSchemeParam is the name of the OpenID4VP client_id_scheme parameter.
+// Specified by https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-authorization-request
+const clientIDSchemeParam = "client_id_scheme"
```
Comment (gerardsn) [inline auth/api/iam/types.go:141]: related to `clientIdSchemesSupported`

File: auth/api/iam/types.go [hunk 5]
```diff
@@ -99,6 +96,58 @@ var vpFormatsSupported = map[string]map[string][]string{
...
+
+// vpTokenParam is the name of the OpenID4VP vp_token parameter.
+// Specified by https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-response-type-vp_token
+const vpTokenParam = "vp_token"
+
+// responseModeParam is the name of the OAuth2 response_mode parameter.
+// Specified by https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html
+const responseModeParam = "response_mode"
```
Comment (gerardsn) [inline auth/api/iam/types.go:149]: related to other `responseMode*` parameters

**Review by gerardsn — COMMENTED** [review]
Mainly a consistency thing.

I'm also wondering how this large group of constants can be somewhat organized. I think grouping the constants of a param key with its values brings some structure to this. So placing `responseTypeParam`, `responseTypesSupported` and the supported types all together?

**Comment by reinkrul** [comment]
> Mainly a consistency thing.
> 
> I'm also wondering how this large group of constants can be somewhat organized. I think grouping the constants of a param key with its values brings some structure to this. So placing `responseTypeParam`, `responseTypesSupported` and the supported types all together?

I now moved them together, a bit better now i.m.o.

**Review by gerardsn — APPROVED** [review]

**Pull request merged**